tests/unit/steps/test_steps_analyze.py

```diff
@@ -55,31 +55,73 @@ async def test_analysis_dump(mocker, apps):
+        "  units:\n"
+        "    rabbitmq-server/0:\n"
+        "      workload_version: '3.8'\n"
+        "      os_version: yoga\n"
     )
 
-    mocker.patch.object(analyze.Analysis, "_populate", return_value=apps)
+    mocker.patch.object(
+        analyze.Analysis,
+        "_populate",
+        return_value=[apps["keystone_ussuri"], apps["cinder_ussuri"], apps["rmq_ussuri"]],
+    )
     result = await analyze.Analysis.create()
     assert str(result) == expected_result
 
 
 @pytest.mark.asyncio
-async def test_generate_model(mocker, full_status, config):
+async def test_populate_model(mocker, full_status, config):
     mocker.patch.object(analyze, "async_get_status", return_value=full_status)
     mocker.patch.object(
         analyze, "async_get_application_config", return_value=config["openstack_ussuri"]
     )
-    # Initially, 3 applications are in the status (keystone, cinder and rabbitmq-server)
-    assert len(full_status.applications) == 3
+    # Initially, 4 applications are in the status: keystone, cinder, rabbitmq-server and my-app
+    # my-app is a unknown application in the model.
+    assert len(full_status.applications) == 4
     apps = await Analysis._populate()
-    assert len(apps) == 3
-    assert {app.charm for app in apps} == {"keystone", "cinder", "rabbitmq-server"}
+    assert len(apps) == 4
+    # apps are on the UPGRADE_ORDER sequence and unknown in the end of the list
+    assert [app.charm for app in apps] == ["rabbitmq-server", "keystone", "cinder", "my-app"]
 
 
 @pytest.mark.asyncio
 async def test_analysis(mocker, apps):
-    """Test analysis function."""
-    expected_result = analyze.Analysis(apps=apps)
-    mocker.patch.object(analyze.Analysis, "_populate", return_value=apps)
+    """Test analysis object."""
+    app_keystone = apps["keystone_ussuri"]
+    app_cinder = apps["cinder_ussuri"]
+    app_rmq = apps["rmq_ussuri"]
+    app_mysql_router = apps["mysql_router_ussuri"]
+    no_openstack = apps["no_openstack"]
+    expected_result = analyze.Analysis(
+        apps=[app_rmq, app_keystone, app_cinder, app_mysql_router, no_openstack]
+    )
+    mocker.patch.object(
+        analyze.Analysis,
+        "_populate",
+        return_value=[app_rmq, app_keystone, app_cinder, app_mysql_router, no_openstack],
+    )
+    mock_logger = mocker.patch("cou.steps.analyze.logger")
 
     result = await Analysis.create()
     assert result == expected_result
+    assert result.current_cloud_os_release == "ussuri"
+    # no OpenStack charms doesn't show on logs.
+    mock_logger.warning.assert_called_once_with(
+        "Disconsidering %s to determine the minimum version of the cloud.", app_mysql_router.name
+    )
+    mock_logger.debug.assert_called_once_with(
+        (
+            "Disconsidering %s to determine the min version of the cloud "
+            "because isn't an OpenStack charm."
+        ),
+        no_openstack.name,
+    )
+    assert result.current_cloud_os_release.next_release == "victoria"
```

Comment:
Isn't this out of scope for this test? Having `.next_release` return the right value should be tested (and is tested) in `test_openstack.py`.
If a regression occurs in `utils/openstack.py` this test may fail and confuse the developer.

Reply:
Done